deeds-dapp-service/src/main/java/io/meeds/deeds/web/rest/AuthorizationCodeController.java

```diff
@@ -65,13 +69,19 @@ public void generateCode(Principal principal,
   @ResponseStatus(value = HttpStatus.NO_CONTENT)
   public void checkValidity(Principal principal,
                             @RequestHeader(name = CODE_VERIFICATION_HTTP_HEADER, required = true)
-                            int code) {
-    if (principal == null || StringUtils.isBlank(principal.getName())) {
+                            int code,
+                            @RequestParam(name = "email")
+                            String email) {
```

Comment:
email not needed, you can store it in current `HTTPSession`

```suggestion
                            int code) {
```